test/merger/test_general_merger.py

```diff
@@ -23,6 +23,33 @@ def test_compose_bloch_sphere_rotations_different_axis() -> None:
     assert composed == BlochSphereRotation(qubit=123, axis=(1, 1, 0), angle=math.pi)
 
 
+@pytest.mark.parametrize(
+    ("bsr_a", "bsr_b", "expected_result"),
+    [
+        (I(0), Rx(0, theta=math.pi), Rx(0, theta=math.pi)),
+        (Rx(0, theta=math.pi), I(0), Rx(0, theta=math.pi)),
+        (X(0), X(0), I(0)),
+        (H(0), H(0), I(0)),
+        (Rx(0, theta=math.pi), Rx(0, theta=math.pi), I(0)),
+        (Rx(0, theta=math.pi / 2), Rx(0, theta=math.pi / 2), Rx(0, theta=math.pi)),
+        (Rx(0, theta=math.pi), Ry(0, theta=math.pi / 2), BlochSphereRotation(0, axis=(1, 0, 1), angle=math.pi)),
+    ],
+    ids=[
+        "[bsr_a == I]",
+        "[bsr_b == I]",
+        "[bsr_a.generator == bsr_b.generator] X + X == I",
+        "[bsr_a.generator == bsr_b.generator] H + H == I",
+        "[bsr_a.generator == bsr_b.generator] Rx(pi) + Rx(pi) == I",
+        "[bsr_a.generator == bsr_b.generator] Rx(pi/2) + Rx(pi/2) != X",
+        "[bsr_a.generator != bsr_b.generator] Rx(pi) + Ry(pi/2) != H",
```

Comment:
In our Slack conversation we indeed used addition to discuss the various cases, but multiplication is more appropriate. 
```suggestion
        "[bsr_a.generator == bsr_b.generator] X * X == I",
        "[bsr_a.generator == bsr_b.generator] H * H == I",
        "[bsr_a.generator == bsr_b.generator] Rx(pi) * Rx(pi) == I",
        "[bsr_a.generator == bsr_b.generator] Rx(pi/2) * Rx(pi/2) = Rx(pi) ~ X",
        "[bsr_a.generator != bsr_b.generator] Rx(pi) * Ry(pi/2) ~ H",
```
The `~` character is used to denote 'equal up to a global phase'.